### ISSUE TITLE
Armor stands can be destroyed by fireworks

### ISSUE THREAD

**Issue by AIphaDragon** [LlmDl/Towny#2854]
We are not sure is it actually connected with Towny, but in towns armor stands can be blown up and destroyed by fireworks. Is there a way to toggle it somehow?
Here is a video: https://www.youtube.com/watch?v=6J6CvzMSFzE

Also, being a russian speaking server, we would like to ask you is there a possibility to translate RESIDENT, ALLY, OUTSIDER and FRIEND groups as well as BUILD, DESTROY, SWITCH, ITEMUSE actions somewhere in code? And where exactly, in what section? We would appreciate your help.

We are looking forward to your answer

Server start: https://pastebin.com/4aVs2GCV
Towny version: 0.92.0.9
config: https://pastebin.com/GiUNSzHg
Townyperms: https://pastebin.com/Gxs0CJDf

**Comment by LlmDl** [comment]
Those words aren't translateable because they're used in commands.

I have a test build I need you to use to try the fix for fireworks. Please read this changelog before you use it, there are config changes you must make.

```
v0.92.0.10:
    - Add support for Reserve courtesy of creatorfromhell.
      - Reserve is an optional Vault-replacement available at https://www.spigotmc.org/resources/reserve.50739/
    - Add /town new to the /town ? help, don't know how this one was missed.
    - Added /town create as an alias of sorts to /town new.
    - Improve error seen when a player tries to use /t set spawn when not in the homeblock or when the homeblock isn't set.
    - Added /ta giveplots as an alias of sorts to /ta givebonus.
    - Language files bumped to 0.44.
    - New Command: '/ta set plot {town}' courtesy of Articdive.
      - When in a town only a single plot can be transfered at one time.
        - Admin command to change a plot from one town to another.
        - Does not require a town to have available townblocks to claim.
      - When in the wilderness two types of sub commands can be used to do area claims:
        - Admin command to change wilderness into a claim for a town.
        - Does require a town to have available townblocks to claim.
        - Does obey proximity rules for claims between towns/homeblocks.
        - /ta set plot {town} {rect|circle} {radius}
        - /ta set plot {town} {rect|circle} auto
      - Does not charge town for the townblock.
      - Requires towny.command.townyadmin.set.plot permission node.
    - New Permission node: towny.command.townyadmin.set.plot
      - Default: false
      - Child node of towny.admin and towny.command.townyadmin.set.*
      - For mods who don't have the full towny.admin permission node but are able to change plots to other towns.
    - New Permission node: towny.command.townyadmin.set.capital
      - Default: false
      - Child node of towny.admin and towny.command.townyadmin.set.*
    - New Permission node: towny.command.townyadmin.set.title
      - Default: false
      - Child node of towny.admin and towny.command.townyadmin.set.*
    - New Permission node: towny.command.townyadmin.set.surname
      - Default: false
      - Child node of towny.admin and towny.command.townyadmin.set.*
    - New Feature: Limiting Outposts by Town and Nation Levels!
      - See below.
    - New Config Option: global_town_settings.limit_outposts_using_town_and_nation_levels
      - default: false
      - When set to true outposts can be limited by the townOutpostLimit value of the Town Levels and
        the nationBonusOutpostLimit value in the Nation Levels. 
      - In this way nations can be made to be the only way of receiving outposts, or as an incentive to receive more outposts.
      - Towns which are larger can have more outposts.
      - When activated, this setting will not cause towns who already have higher than their limit to lose outposts. 
      - They will not be able to start new outposts until they have unclaimed outposts to become under their limit. 
      - Likewise, towns that join a nation and receive bonus outposts will be over their limit if they leave the nation.
      - When the feature is enabled, the limits are shown in the /town status screen.
    - New Config Option: global_townSettings.over_outpost_limits_stops_teleports
      - default: false
      - When limit_outposts_using_town_and_nation_levels is also true, towns which are over their outpost
        limit will not be able to use their /town outpost teleports for the outpost #'s higher than their limit, 
        until they have dropped below their limit.
      - eg: If their limit is 3 then they cannot use /t outpost 4 
    - REQUIRED CONFIG CHANGE:
      - a new entry to all of your nation_levels in the config must be added, otherwise Towny will not start up properly.
      - add 'nationBonusOutpostLimit: #' to each nation_level.
      - default settings are between 0 and 3.
      - If you do not want to do this you can delete the nation_level section of your config and Towny will re-create it with the missing lines and new default settings.
    - REQUIRED CONFIG CHANGE:
      - a new entry to all of your town_levels in the config must be added, otherwise Towny will not start up properly.
      - add 'townOutpostLimit: #' to each town_level.
      - default settings are between 0 and 4.
      - If you do not want to do this you can delete the town_level section of your config and Towny will re-create it with the missing lines and new default settings.
    - New Config Option: town.min_distance_for_outpost_from_plot
      - default: 5
      - Minimum number of plots an outpost must be from any other town's plots.
      - Useful when min_plot_distance_from_town_plot is set to near-zero to allow towns to have claims 
        near to each other, but want to keep outposts away from towns. 
    - 65+ new languages strings are now translateabl. 
      - These strings are mainly found in the /plot perm, /res, /town, /nation and /townyworld status screens.
    - New Config Option: global_town_settings.homeblocks_prevent_forcepvp
      - default: false
      - If set to true, when a world has forcepvp set to true, homeblocks of towns will not be affected and have PVP set to off.
    - New Feature: /townyworld commands can be run from console!
      - Requires the worldname to be supplied after 'tw'
      - Eg: 'tw world_nether toggle forcepvp'
      - /tw regen and /tw undo are not usable from the console because they require a player standing in a plot.
    - New Feature: /town, /res and /nation can be run from the console!
      - Commands are limited but generally /{command} list and /{command} {name} work.
      - Previously the console would just show the help page for the /{command} no matter what sub command was used. 
    - REQUIRED CONFIG CHANGE: add END_CRYSTAL to the protection.item_use_ids list.
      - Stops end crystals being placed in towns where they shouldn't be able to place items. 
    - Fix protection of End Crystals in town.
```
https://www.dropbox.com/s/fomct5bern2vo1z/Towny%20AlphaDragon.jar?dl=0

Let me know if that fixes it.

**Comment by AIphaDragon** [comment]
Unfortunately, it didn't help.
Also, we have another problem with Sheriff permission ("Шериф" in our Townyperms.yml)

We gave him "towny.command.plot.set.*" permission, but it didn't work. Then we decided to specify it as "towny.command.plot.set.perm" and so on for every command, but it still doesn't work. Our Townyperms.yml is attached to this ticket.
How can we solve this problem?
Thank you in advance

**Comment by LlmDl** [comment]
RE: the fireworks, what item exactly are they using and how do I give myself that item. I had a problem before with firework types.

RE: the plot thing, what command is the sheriff trying to use?

**Comment by AIphaDragon** [comment]
1) To destroy the armor stand, you need to fire explosive fireworks. Craft: https://imgur.com/a/9qGvB.
2) All commands "/plot set".

**Comment by LlmDl** [comment]
For the sheriff rank:

Consider adding this permission node: 
- towny.command.plot.asmayor – For town plot management, grant this to any ranks you want to have the ability to:
  - reclaim plots from players for the town.
  - Toggle perms and plot settings on any plot in the town.
  - Put plots up for sale and take them down again.
  - Mayors and Assistants can still set plots for sale without the node.

**Comment by LlmDl** [comment]
The fireworks are fired from dispensers? Should I be able to fire them by hand?

**Comment by LlmDl** [comment]
I figured it out. Is fixed as of 0.92.0.10. Note that explosions have to be false for them to be protected.
